File: 0125-valid-palindrome/0125-valid-palindrome.ts
function isPalindrome(s: string): boolean {
    const sp: string[] = s.split('');
    const fil = sp.filter(c=>/[a-zA-Z0-9]/.test(c));
    const data = fil.map(f=>f.toLowerCase());
 
    let i = 0;
    let j = data.length - 1;
    // console.log(data);
    while(i<j){
        // console.log(i,j,data[i], data[j]);
        if(data[i++] !== data[j--]){
            return false;
        }
    }
    return true;
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = list(filter(lambda c: (c>='A' and c<='Z') or (c>='a' and c<='z') or (c>='0' and c<='9'),s));
        print(s)
        
        s = list(map(lambda c: c.lower(), s))
        
        i = 0
        j = len(s) - 1
        while i < j:
            if s[i] != s[j]:
                return False
            i+=1
            j-=1
            
        return True